### DISCUSSION TITLE
AddHostObjectToScript Usage

### DISCUSSION THREAD

**Discussion by drearyrainDeng** [MicrosoftEdge/WebView2Feedback#4108] [Q&A]
Follow the guide, i add 'bridge' object to web side.  call Fun(), after set 'Prop',  it print the old prop value , Is this the design or is it the wrong way I set the properties? If I keep it running in sync, how should I call it?

![addhostobjecttoscript](https://github.com/MicrosoftEdge/WebView2Feedback/assets/35326823/1f4cb7a8-2373-4c1a-9410-7d9652c0ed72)

**Comment by victorhuangwq** [comment]
@vbryh-msft could you take a look at this question?

**Comment by vbryh-msft** [comment]
You are doing everything right. But there are multiple processes involved - to avoid any races you should do await after setting the property to be sure that it is set, or use `chrome.webview.hostObjects.sync`  - but it will be blocking.

**Reply by drearyrainDeng** [reply]
yes, `chrome.webview.hostObjects.sync.bridge.AnotherObject.Prop  = "new proValue"` is work. do you mean "do await after setting the property" like that `bridge.AnotherObject.Prop = "new proValue"; await bridge.AnotherObject.Prop `? (An additional line needs to be added to ensure that the property had updated).
